kaniko.go

```diff
@@ -174,6 +182,47 @@ func (p Plugin) Exec() error {
 		return fmt.Errorf("repository name to publish image must be specified")
 	}
 
+	if p.Build.PushOnly {
+		// When push-only is set, source_tar_path MUST be provided
+		if p.Build.SourceTarPath == "" {
+			return fmt.Errorf("source_tar_path is required when push_only is set. Please provide a valid tarball path.")
+		}
+
+		if _, err := os.Stat(p.Build.SourceTarPath); os.IsNotExist(err) {
+			return fmt.Errorf("image tarball does not exist at path: %s", p.Build.SourceTarPath)
+		}
+
+		if p.Build.Repo == "" {
+			return fmt.Errorf("destination repository name must be specified when push-only is set")
+		}
+
+		// Load the image from the tarball
+		img, err := crane.Load(p.Build.SourceTarPath)
+		if err != nil {
+			return fmt.Errorf("failed to load image from tarball: %v", err)
+		}
+
+		// If no tags are specified, use 'latest'
+		tags := []string{"latest"}
+		if len(p.Build.Tags) > 0 {
+			tags = p.Build.Tags
+		}
```

Comment:
```suggestion
		tags := p.Build.Tags
```

this is redundant as in main.go it is already ensured that the default value for `tags` is `latest` via the CLI flag